engine/battle/core.asm

```diff
@@ -2557,11 +2557,12 @@ MoveSelectionMenu:
 	ld a, [wLinkState]
 	cp LINK_STATE_BATTLING
 	jr z, .matchedkeyspicked
+	; Allow the buttons specified below in regular battles.
 	ld a, [wFlags_D733]
 	bit BIT_TEST_BATTLE, a
 	ld b, D_UP | D_DOWN | A_BUTTON | B_BUTTON | SELECT
 	jr z, .matchedkeyspicked
-	ld b, $ff
+	ld b, -1
```

Comment:
```suggestion
	ld b, D_UP | D_DOWN | D_LEFT | D_RIGHT | A_BUTTON | B_BUTTON | SELECT | START
```

Reply:
It might be worth adding `BUTTONS` and `D_PAD` constants like pokecrystal has for this sort of large combined mask.